powerauth-data-adapter/src/main/java/io/getlime/security/powerauth/lib/dataadapter/impl/service/DataAdapterService.java

```diff
@@ -0,0 +1,272 @@
+package io.getlime.security.powerauth.lib.dataadapter.impl.service;
+
+import io.getlime.security.powerauth.crypto.server.util.DataDigest;
+import io.getlime.security.powerauth.lib.dataadapter.api.DataAdapter;
+import io.getlime.security.powerauth.lib.dataadapter.configuration.DataAdapterConfiguration;
+import io.getlime.security.powerauth.lib.dataadapter.exception.AuthenticationFailedException;
+import io.getlime.security.powerauth.lib.dataadapter.exception.SMSAuthorizationFailedException;
+import io.getlime.security.powerauth.lib.dataadapter.impl.validation.AuthenticationRequestValidator;
+import io.getlime.security.powerauth.lib.dataadapter.impl.validation.CreateSMSAuthorizationRequestValidator;
+import io.getlime.security.powerauth.lib.dataadapter.model.entity.AuthMethodChoiceEntity;
+import io.getlime.security.powerauth.lib.dataadapter.model.entity.BankAccountChoiceEntity;
+import io.getlime.security.powerauth.lib.dataadapter.model.entity.BankAccountEntity;
+import io.getlime.security.powerauth.lib.dataadapter.model.entity.FormDataChangeEntity;
+import io.getlime.security.powerauth.lib.dataadapter.model.request.*;
+import io.getlime.security.powerauth.lib.dataadapter.model.response.AuthenticationResponse;
+import io.getlime.security.powerauth.lib.dataadapter.model.response.BankAccountListResponse;
+import io.getlime.security.powerauth.lib.dataadapter.model.response.CreateSMSAuthorizationResponse;
+import io.getlime.security.powerauth.lib.dataadapter.model.response.UserDetailResponse;
+import io.getlime.security.powerauth.lib.dataadapter.repository.SMSAuthorizationRepository;
+import io.getlime.security.powerauth.lib.dataadapter.repository.model.entity.SMSAuthorizationEntity;
+import io.getlime.security.powerauth.lib.nextstep.model.entity.OperationAmountAttribute;
+import org.joda.time.DateTime;
+import org.springframework.context.MessageSource;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.support.ReloadableResourceBundleMessageSource;
+import org.springframework.core.MethodParameter;
+import org.springframework.stereotype.Service;
+import org.springframework.validation.BeanPropertyBindingResult;
+import org.springframework.validation.ValidationUtils;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+
+import java.math.BigDecimal;
+import java.util.*;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Sample implementation of DataAdapter interface which should be updated in real implementation.
+ *
+ * @author Roman Strobl
+ */
+@Service
+public class DataAdapterService implements DataAdapter {
```

Comment:
Not nice. You have basically just moved the code from one class to another - this does not make much sense. The idea was to extract the **custom logic** into separate service, not the whole code. Simply put, the code that should be extracted should only implement the logic that is required for customization. For example, DataAdapterService must not contain SMSAuthorizationRepository, because the customizing client knows nothing about how it works. Instead, he should have a method:

`boolean sendSms(user, message)`

... that just does the job using the stuff he know how works. Etc.

Since the issue is related mainly to the SMS sending, I will probably accept this pull request and then we will figure out how to improve the code.